internal/workflow.go

```diff
@@ -513,6 +513,37 @@ func (wc *workflowEnvironmentInterceptor) AwaitWithTimeout(ctx Context, timeout
 	return true, nil
 }
 
+// AwaitWithTimeoutAndOptions blocks the calling thread until condition() returns true
+// Returns ok equals to false if timed out and err equals to CanceledError if the ctx is canceled.
+func AwaitWithTimeoutAndOptions(ctx Context, timeout time.Duration, options TimerOptions, condition func() bool) (ok bool, err error) {
+	assertNotInReadOnlyState(ctx)
+	state := getState(ctx)
+	return state.dispatcher.interceptor.AwaitWithTimeoutAndOptions(ctx, timeout, options, condition)
+}
+
+func (wc *workflowEnvironmentInterceptor) AwaitWithTimeoutAndOptions(ctx Context, timeout time.Duration, options TimerOptions, condition func() bool) (ok bool, err error) {
+	state := getState(ctx)
+	defer state.unblocked()
+	if options.Summary == "" {
```

Comment:
not sure why someone might call `AwaitWithTimeoutAndOptions` without specifying a summary, but defaulting to "AwaitWithTimeoutAndOptions"

Reply:
I'm 50/50 here. We do not default a summary in `NewTimerWithOptions`, not sure we need to here. But then again it makes some sense since we are in the option-less equivalent calls. No strong opinion here.

Reply:
After some thought, i think I'm leaning the other way now, if someone chooses to use `AwaitWithTimeoutAndOptions`, they should probably be responsible for setting their `TimerOptions` properly

Reply:
Yeah I'd be consistent with `NewTimerWithOptions` . We should set some sensible default if the user is calling `AwaitWithTimeout`